vowpalwabbit/cb_explore.cc

```diff
@@ -235,6 +236,7 @@ void predict_or_learn_cover(cb_explore& data, single_learner& base, example& ec)
     }
   }
 
+  ec.l.cs.costs = v_init<COST_SENSITIVE::wclass>();
```

Comment:
so basically you have to clear the costs of any label that was temporarily used by a reduction? if so you think its possible that we're missing this clear on other reductions? or how would you explain the need for this line

Reply:
There are some reductions which do a by value copy of the entire label and since v_arrays shallow copy currently this can result in double frees.

So you only need to clear if something else made a shallow copy of it. In future (as soon as this is merged) we will use move/copy to make this more sane.

Reply:
is this also related to the double free with nn?

Reply:
No this was a different one. Caused by the fact that we reuse the label in the reduction data, hence the need to clear it so that at tear down there is only one reference left.

It boils down to at destruction time either only the example or only the reduction data needs the reference and not both

Reply:
so the reduction calls `delete_v` on the array?

Reply:
Yep the example will call delete_v on the array in the delete unions function